QuadradiusR/scripts/game_manager.gd

```diff
@@ -13,16 +13,28 @@ func _ready():
 # # # # # # # # #
 func _game_state_diff(data: Dictionary):
 	var diff = data['game_state_diff']
-	if diff['current_player_id'] == Context.user_id:
-		# Play opponent's move
-		if 'pieces' in diff['board']:
-			for piece_id in diff['board']['pieces'].keys():
-				if piece_id[0] == "$":
-					continue
-				var dest_tile_id = diff['board']['pieces'][piece_id]['tile_id']
-				var source_tile_pos = Context.game_state.get_tile_pos_by_piece_id(piece_id)
-				var dest_tile_pos = Context.game_state.get_tile_pos_by_tile_id(dest_tile_id)
-				board.move_torus_by_tiles(source_tile_pos, dest_tile_pos)
+	if 'pieces' in diff['board']:
+		for piece_id in diff['board']['pieces'].keys():
+			if piece_id[0] == "$":
+				continue
+			
+			var piece = diff['board']['pieces'][piece_id]
+			if board.get_torus_by_id(piece_id).player_id == Context.user_id:
+				continue
+			
+			var source_tile_pos = Context.game_state.get_tile_pos_by_piece_id(piece_id)
+			var dest_tile_pos = Context.game_state.get_tile_pos_by_tile_id(piece['tile_id'])
+			
+			board.move_torus_by_tiles(source_tile_pos, dest_tile_pos)
```

Comment:
I think each `if` should be a function so that `_game_state_diff` is not that long and hard to understand 😃

Reply:
I've noticed this too and fixed it in next issue I'm working on, but I'll add it here too
- [ ] Refactor gamestate parsing